site/content/docs/01-component-format.md

```diff
@@ -189,6 +172,53 @@ Local variables (that do not represent store values) must *not* have a `$` prefi
 </script>
 ```
 
+---
+
+You can create your own stores without relying on [`svelte/store`](docs#svelte_store), by implementing the **store contract**:
+
+1) A store must contain a `.subscribe` method, which must accept as its argument a subscription function. This subscription function must be immediately and synchronously called with the store's current value upon calling `.subscribe`. All of a store's active subscription functions must later be synchronously called whenever the store's value changes.
+2) The `.subscribe` method must return an unsubscribe function. Calling an unsubscribe function must stop its subscription, and its corresponding subscription function must not be called again by the store.
+3) A store may *optionally* contain a `.set` method, which must accept as its argument a new value for the store, and which synchronously calls all of the store's active subscription functions. Such a store is called a *writable store*.
+
+```js
+
+/* Example: a custom writable store for `location.hash` */
+
+const subscriptions = [];
+
+let lastHash;
```

Comment:
What specifically is this trying to address? I think the built-in writable stores will call subscribers again if you `.set` them to their current value, so it might be a bit confusing if this custom store didn't.

Reply:
Setting `location.hash` triggers the `hashchange` event asynchronously, so without this check every `.set()` would call the subscribers twice.